sky/cli.py

```diff
@@ -3471,11 +3474,19 @@ def storage_delete(names: List[str], all: bool):  # pylint: disable=redefined-bu
     if sum([len(names) > 0, all]) != 1:
         raise click.UsageError('Either --all or a name must be specified.')
     if all:
-        click.echo('Deleting all storage objects.')
         storages = sky.storage_ls()
         names = [s['name'] for s in storages]
     else:
         names = _get_glob_storages(names)
+    if not yes and len(names) > 0:
```

Comment:
Some edge case to keep in mind while resolving the comment above is to make sure to handle the case when multiple storage names are given, but some of them do not exist. For example, when you have following storages, and when `sky storage delete doyoung-r2-test storage-not-exsit` is ran, we want to make sure to inform that `storage-not-exist` does not exist and prompt the user for deleting the existing ones, which in this case is `doyoung-r2-test`. This is currently handled well by `_get_glob_storages`, but still wanted to give some heads up.

```
$ sky storage ls
NAME                            UPDATED      STORE      COMMAND                       STATUS  
doyoung-r2-test                 12 mins ago  R2         sky launch yaml/r2_s3_sto...  READY   
doyoung-s3-test                 12 mins ago  S3         sky launch yaml/r2_s3_sto...  READY
$ sky storage delete doyoung-r2-test storage-not-exist
Storage storage-not-exist not found.
Deleting 1 storage: doyoung-r2-test. Proceed? [Y/n]:
```